modules/minimongo-observable/observable-cursor.ts

```diff
@@ -79,13 +80,31 @@ export class ObservableCursor<T> extends Observable<T[]> {
     });
   }
 
+  addedAt(doc, at, before) {
+    this.data.splice(at, 0, doc);
+    this.handleChange();
+  }
+
+  changedAt(doc, old, at) {
+    this.data[at] = doc;
```

Comment:
Why do you think the fetch is bad in this case?
I think the fetch just iterates and returns a rep of the internal collection (https://github.com/meteor/meteor/blob/abd574f38008b45f5e2a6bc322b10bcdde44763a/packages/minimongo/minimongo.js#L207). Keeping own data and using the fetch has near the same performance, at the same time it's much simpler.